MoreLinq.Test/PartitionTest.cs

```diff
@@ -0,0 +1,194 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008 Jonathan Skeet. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System.Linq;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    using System;
+
+    [TestFixture]
+    public class PartitionTest
+    {
+        [Test]
+        public void PartitionWithResultSelector()
+        {
+            var (evens, odds) =
+                Enumerable.Range(0, 10)
+                          .Partition(x => x % 2 == 0, ValueTuple.Create);
```

Comment:
This is a very nice example of this overload. Maybe it should be included in the xmldoc?